File: 🟢 Easy/145 - Binary Tree Postorder Traversal/_145BinaryTreePostorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public static void postOrder(TreeNode root, ArrayList<Integer> arr) {
        if (root == null) {
            return;
        }
        postOrder(root.left, arr);
        postOrder(root.right, arr);
        arr.add(root.val);
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> al = new ArrayList<>();
        postOrder(root, al);
        return al;
    }
}

File: 🟢 Easy/145 - Binary Tree Postorder Traversal/_145BinaryTreePostorderTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

var postOrder = function (root, arr) {
    if (root == null) {
        return;
    }
    postOrder(root.left, arr);
    postOrder(root.right, arr);
    arr.push(root.val);
}

var postorderTraversal = function (root) {
    let arr = [];
    postOrder(root, arr);
    return arr;
};